newIDE/app/src/EventsSheet/EventsTree/DropContainer.js

```diff
@@ -0,0 +1,295 @@
+// @flow
+import * as React from 'react';
+import { getIndentWidth, type SortableTreeNode } from '.';
+import {
+  moveNodeAbove,
+  moveNodeBelow,
+  moveNodeAsSubevent,
+  isSameDepthAndBelow,
+} from './helpers';
+import { type WidthType } from '../../UI/Reponsive/ResponsiveWindowMeasurer';
+import './style.css';
+import GDevelopThemeContext from '../../UI/Theme/ThemeContext';
+
+const styles = {
+  dropArea: { zIndex: 1, position: 'absolute' },
+  dropIndicator: {
+    position: 'absolute',
+    zIndex: 2,
+    border: '4px solid black',
+    outline: '1px solid white',
+  },
+};
+
+const getTargetPositionStyles = (
+  indentWidth: number,
+  draggedNodeHeight: number,
+  isDraggedNodeChild: boolean
+) => ({
+  'bottom-left': { left: '0px', bottom: '0px', top: '50%', width: indentWidth },
+  'bottom-right': {
+    left: `${indentWidth}px`,
+    right: '0px',
+    bottom: '0px',
+    top: '50%',
+  },
+  top: { left: '0px', right: '0px', top: '0px', bottom: '50%' },
+  bottom: { left: '0px', right: '0px', top: '50%', bottom: '0px' },
+  'bottom-bottom-left': {
+    left: isDraggedNodeChild ? '0px' : '10px',
+    top: '100%',
+    height: draggedNodeHeight,
+    width: indentWidth,
+  },
+  'bottom-bottom-right': {
+    left: isDraggedNodeChild ? `${indentWidth + 10}px` : `${indentWidth}px`,
+    right: '0px',
+    top: '100%',
+    height: draggedNodeHeight,
+  },
+});
+
+const getIndicatorPositionStyles = (indentWidth: number) => ({
+  bottom: { left: '0px', right: '0px', bottom: '-4px' },
+  'bottom-right': { left: `${indentWidth}px`, right: '0px', bottom: '-4px' },
+  top: { left: '0px', right: '0px', top: '-4px' },
+});
+
+type DropTargetContainerStyle = {|
+  left?: string,
+  right?: string,
+  top?: string,
+  bottom?: string,
+  width?: number,
+  height?: number,
+|};
+
+function DropTargetContainer({
+  DnDComponent,
+  canDrop,
+  onDrop,
+  style,
+}: {|
+  DnDComponent: any,
+  canDrop: () => boolean,
+  onDrop: () => void,
+  style: {
+    dropIndicator: DropTargetContainerStyle,
+    dropArea: DropTargetContainerStyle,
+  },
+|}) {
+  const gdevelopTheme = React.useContext(GDevelopThemeContext);
+  return (
+    <DnDComponent canDrop={canDrop} drop={onDrop}>
+      {({ isOver, connectDropTarget, canDrop }) => {
+        return connectDropTarget(
+          <div>
+            {/* Drop area */}
+            <div
+              style={{
+                ...styles.dropArea,
+                ...style.dropArea,
```

Comment:
It's a bit confusing to see theses 2 drop area styles. Can you maybe rename `styles` with `baseStyles` or `sharedStyles`?